python/ray/util/collective/collective_group/nccl_util.py

```diff
@@ -194,6 +200,25 @@ def get_tensor_strides(tensor):
                      "cupy.ndarray.".format(type(tensor)))
 
 
+def get_tensor_device(tensor):
+    """Return the GPU index of a tensor."""
+    if isinstance(tensor, cupy.ndarray):
+        try:
+            device = tensor.device.id
+        except AttributeError as e:
+            raise RuntimeError("Got an exception {}. The tensor is not on "
+                               "a valid GPU.".format(e))
+    elif torch_available():
+        if isinstance(tensor, torch.Tensor):
+            device = tensor.device.index
+            if not isinstance(device, int):
+                raise RuntimeError("The tensor is not on a valid GPU.")
```

Comment:
I think you need to merge the tensor and torch_available check (or else you have a unbounded local variable error)